src/MongoDB.Driver/MongoCredential.cs

```diff
@@ -524,6 +536,30 @@ private static MongoCredential FromComponents(string mechanism, string source, s
                         mechanism,
                         new MongoInternalIdentity(source, username),
                         evidence);
+                case "MONGODB-AWS":
+                    // MUST be "$external". Defaults to $external.
+                    EnsureNullOrExternalSource(mechanism, source);
+                    if (username == null)
+                    {
+                        if (evidence != null && evidence is PasswordEvidence)
+                        {
+                            throw new ArgumentException("A MONGODB-AWS must have access key id.");
```

Comment:
I think in order for this sentence to be grammatically correct (and following the pattern used for PLAIN credentials), this might be better phrased as:
"A MONGODB-AWS **credential** must have an access key id."

Reply:
See this comment: https://github.com/MikalaiMazurenka/mongo-csharp-driver/pull/22#discussion_r394724786

Reply:
>See this comment: https://github.com/MikalaiMazurenka/mongo-csharp-driver/pull/22#discussion_r394724786

https://github.com/MikalaiMazurenka/mongo-csharp-driver/pull/22#discussion_r394724786:

>I agree that ArgumentException is questionable. ArgumentException is usually reserved for problems with the arguments of the method itself.
>Not sure InvalidDataException is right either (there is no data...).
>Normally I think we would throw an InvalidOperationException here.

I found out I actually fixed that in one of commits.
Done.